.circleci/config.yml

```diff
@@ -37,10 +72,147 @@ jobs:
+          at: ~/repo
+      - run:
+          name: Deploy Adhoc
+          command: 'make publish-docs-adhoc FLAGS=--forceUpload'
+      # TODO: add a probe step here to validate?
+
+  push-head-docs:
+    executor: default_build_environment
+    working_directory: ~/repo
+
+    steps:
+      - checkout
+      - bootstrap_node
+      - bootstrap_aws_cli
+      - attach_workspace:
+          at: ~/repo
+      - run:
+          name: Deploy Head
+          command: 'make publish-docs-head FLAGS=--forceUpload'
+      # TODO: add a probe step here to validate?
+  
+  push-staging-docs:
+    executor: default_build_environment
+    working_directory: ~/repo
+
+    steps:
+      - checkout
+      - bootstrap_node
+      - bootstrap_aws_cli
+      - attach_workspace:
+          at: ~/repo
+      - run:
+          name: Deploy Staging
+          command: 'make publish-docs-staging FLAGS=--forceUpload'
+      # TODO: add a probe step here to validate?
+  
+  push-prod-docs:
+    executor: default_build_environment
+    working_directory: ~/repo
+
+    steps:
+      - checkout
+      - bootstrap_node
+      - bootstrap_aws_cli
+      - attach_workspace:
+          at: ~/repo
+      - run:
+          name: Deploy Prod
+          command: 'make publish-docs-prod FLAGS=--forceUpload'
+      # TODO: add a probe step here to validate?
+
 workflows:
   version: 2
   commit_validation:
     jobs:
       - build:
           context:
             - dockerhub
+
+  deploy:
+    jobs:
+      - build:
+          context:
+            - dockerhub
+          filters:
+            branches:
+              only:
+                - release
+      - utils/cancel-older-awaiting-approvals:
+          context:
+            - dockerhub
+            - util
+          circle_token: ${CIRCLECI_TOKEN}
+          requires:
+            - build
+      - build-docs:
+          context:
+            - dockerhub
+          requires:
+            - build      
+      # TODO(spencer): figure out why we aren't serving CDN on adhoc, technically this step isn't needed
+      # as long as that is true
+      - push-adhoc-docs:
+          context:
+            - dockerhub
+          requires:
+            - build-docs
+      - push-head-docs:
+          context:
+            - dockerhub
+          requires:
+            - push-adhoc-docs
+      - push-staging-docs:
+          context:
+            - dockerhub
+          requires:
+            - push-head-docs
+      - utils/slack-notify-waiting-for-approval:
+          slack_bot_token: ${PUSH_REMINDER_BOT_TOKEN}
+          requires:
+            - push-staging-docs
+          context:
+            - util
+      - wait-for-approval:
+          context:
+            - dockerhub
+          type: approval
+          requires:
+            - push-staging-docs
+      - push-prod-docs:
+          context:
+            - dockerhub
+          requires:
+            - wait-for-approval
+      # TODO: add publish NPM package with approval?
```

Comment:
ah, good idea. so eventually we can leverage this to publish to npm? like the direction.